portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerImpl.java

```diff
@@ -83,29 +80,31 @@ private void pollForJob() {
     }
 
     private void processJob(UUID jobId, PortabilityJob job) {
-        PortableDataType dataType = PortableDataType.valueOf(job.transferDataType());
         try {
-            Crypter decrypter = CrypterFactory.create(workerJobMetadata.getKeyPair().getPrivate());
+            Decrypter decrypter = DecrypterFactory.create(workerJobMetadata.getKeyPair().getPrivate());
             JobAuthorization jobAuthorization = job.jobAuthorization();
             String serializedExportAuthData =
                     decrypter.decrypt(jobAuthorization.encryptedExportAuthData());
             AuthData exportAuthData = deSerialize(serializedExportAuthData);
             String serializedImportAuthData =
                     decrypter.decrypt(jobAuthorization.encryptedImportAuthData());
             AuthData importAuthData = deSerialize(serializedImportAuthData);
-            PortabilityCopier
-                    .copyDataType(registry, dataType, job.exportService(), exportAuthData,
-                            job.importService(), importAuthData, jobId);
+            // TODO: Refactor copy logic
+            InMemoryTransferCopier copier = new PortabilityInMemoryTransferCopier(registry);
```

Comment:
Can copier be injected? (once its ctor is annotated with @Inject)

Reply:
Yes, the TODO should account for that.